### PR TITLE
Test: followups to #27823

### LINKED ISSUE #28603: Extend feature_init.py file perturbations
### Motivation

`feature_init.py` is a functional test that perturbs various files in the data directory (simulation data corruption), and checks the response during startup.

Recently, this test was improved in #27823, but there were some open suggestions in, in particular https://github.com/bitcoin/bitcoin/pull/27823#discussion_r1267289758, to extend and/or randomize the way files are perturbed.

There is also the possibility that more sophisticated perturbations could potentially uncover other unhandled exceptions/assert crashes like the one fixed in #27823.  

### Possible solution

_No response_

### Useful Skills

* Python
* Debugging the bitcoin core C++ file parsing code in order to understand why certain perturbation lead to certain errors.

### Guidance for new contributors

Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by L0laL33tz**
Fixes #28603

Added suggested simplifications and implemented randomization

### CODE REVIEW

File: test/functional/feature_init.py [hunk 1]
```diff
@@ -137,8 +138,8 @@ def check_clean_start():
                     # Since the genesis block is not checked by -checkblocks, the
                     # perturbation window must be chosen such that a higher block
                     # in blk*.dat is affected.
-                    tf.seek(150)
-                    tf.write(b'1' * 200)
+                    tf.seek(randint (150, 15000))
```
Comment (brunoerg) [inline test/functional/feature_init.py:141]: In 5ab6419f380cc0a8cde78b125f3eeee5fcba43ae: What is the rationale of the range 150~15000?

Comment (L0laL33tz) [reply test/functional/feature_init.py:141]: I tried to pick a range large enough for randomization to be efficient, but the range itself is arbitrary and could be adjusted.

**Comment by fjahr** [comment]
Concept ACK, better if someone else does a full review though since part of the changes are from me

File: test/functional/feature_init.py (hunk 1 above)
Comment (L0laL33tz) [reply test/functional/feature_init.py:141]: For more clarification @brunoerg the 150 starting point was chosen in the previous test to exclude the genesis block iiuc. The 15000 range should be large enough to provide efficient randomization but small enough to not end outside of the blockfile (200 blocks in the test with ~160 characters each). Could also shorten or enlarge the range ((200*160 = 32000 characters) but I thought 15000 would a good middle ground. 

Let me know if you have any suggestions. Can also add a line of documentation if necessary.

**Comment by achow101** [comment]
ACK 5ab6419f380cc0a8cde78b125f3eeee5fcba43ae

**Review by mzumsande — COMMENTED** [review]
I ran this locally a couple 100 times and  the test failed for me maybe 1% of the time - haven't done any deeper analysis though.

**Pull request merged**